app/src/server/utils/prepareDatasetEntriesForImport.ts

```diff
@@ -54,10 +60,15 @@ export const prepareDatasetEntriesForImport = async (
     return {
       id: uuidv4(),
       datasetId: datasetId,
-      messages: row.input.messages as object[],
+      messages: convertMessages(row.input.messages) as object[],
       function_call: row.input.function_call as object,
       functions: row.input.functions as object[],
-      output: (row.output as unknown as Prisma.InputJsonValue) ?? {
+      tool_choice:
+        row.input.tool_choice || (convertFunctionCall(row.input.function_call) as object),
```

Comment:
I don't like this casting; why aren't we getting the right types from `convertFunctionCall`? Are we sure this is always an object? Couldn't a message have neither `tool_choice` or `function_call`, in which case it would be undefined or null? We have control over the output of this function; we should make sure we do the correct typechecks if necessary and that the types are right without having to cast.

Same thing on lines 63 and 68.

Reply:
It's not always an object. It could be undefined or a string as well. But we need to cast it to something that prisma is willing to save in the tool_choice field, same as we were doing with function_call and functions. If it's actually null and we cast it to an object, that doesn't affect the saving process.